Content.Server/Supermatter/Components/SupermatterComponent.cs

```diff
@@ -0,0 +1,167 @@
+        public const float  EventPowerPenaltyThreshold = 4500f;
+        //The cutoff on power properly doing damage, pulling shit around, and delamming into a tesla. Low chance of pyro anomalies, +2 bolts of electricity
+        public const float  PowerPenaltyThreshold = 5000f;
+        //+1 bolt of electricity, allows for gravitational anomalies, and higher chances of pyro anomalies
+        public const float  SeverePowerPenaltyThreshold = 7000f;
+        //+1 bolt of electricity.
+        public const float  CriticalPowerPenaltyThreshold = 9000f;
+
+        //---------------------------------------------------------------------------------------\\
+
+        //Higher == Crystal safe operational temperature is higher.
+        public const float  HeatPenaltyThreshold = 40f;
+        //is multiplied by ExplosionPoint to cap evironmental damage per cycle
+        public const float  DamageHardcap = 0.002f;
+        //environmental damage is scaled by this
+        public const float  DamageIncreaseMultiplier = 0.25f;
+        //if spaced sm wont take more than 2 damage per cycle
+        public const float MaxSpaceExposureDamage = 2;
+
+        //---------------------------------------------------------------------------------------\\
+
+        //Higher == more overall power
+        public const float  ReactionPowerModefier = 0.55f;
+
+        //---------------------------------------------------------------------------------------\\
+
+        //These would be what you would get at point blank, decreases with distance
+        public const float  DetonationRads = 200f;
+        public const float  DetonationHallucination = 600f;
+
+        //---------------------------------------------------------------------------------------\\
+
+        [ViewVariables(VVAccess.ReadOnly)]
+        public float[] GasComp =
+        {
+            0, //oxy
+            0, //Nit
+            0, //Co2
+            0, //pla
+            0, //tri
+            0, //h2o
+        };
+
+        public readonly float[,] gasFacts =
+        {
+            //Gas Transmit Modifier,                Gas Heat Penality,                Gas Power Mix Ratio
+            {Atmospherics.OxygenTransmitModifier,   Atmospherics.OxygenHeatPenalty,   Atmospherics.OxygenPowerMixRatio},   //oxygen
+            {Atmospherics.NitrogenTransmitModifier, Atmospherics.NitrogenHeatPenalty, Atmospherics.NitrogenPowerMixRatio}, //nitrogen
+            {Atmospherics.CO2TransmitModifier,      Atmospherics.CO2HeatPenalty,      Atmospherics.CO2PowerMixRatio},      //carbon dioxide
+            {Atmospherics.PlasmaTransmitModifier,   Atmospherics.PlasmaHeatPenalty,   Atmospherics.PlasmaPowerMixRatio},   //plasma
+            {Atmospherics.TritiumTransmitModifier,  Atmospherics.TritiumHeatPenalty,  Atmospherics.TritiumPowerMixRatio},  //tritium
+            {Atmospherics.WaterTransmitModifier,    Atmospherics.WaterHeatPenalty,    Atmospherics.WaterPowerMixRatio},    //water
+        };
```

Comment:
Change this to a map for of touples (or make it a struct) for each `Gas` and make it a `DataField` so you can configure it in the prototype. A map with `Gas` as keys is also easier to look up.

Reply:
this should really just be on the gas prototype